LibSWBF2/Wrappers/Terrain.cpp

```diff
@@ -236,8 +300,65 @@ namespace LibSWBF2::Wrappers
 		return p_Terrain->p_Name->m_Text;
 	}
 
+
+	void Terrain::GetHeightMap(uint32_t& dim, uint32_t& dimScale, float_t*& heightData) const
+    {
+        auto info = p_Terrain -> p_Info;
+
+        float_t gridSize     = (float_t) info -> m_GridSize;
+		float_t gridUnitSize = (float_t) info -> m_GridUnitSize;
+
+		float_t maxY = info -> m_HeightCeiling;
+       	float_t minY = info -> m_HeightFloor;
+
+       	float_t halfLength = gridSize * gridUnitSize / 2.0f;
+       	float_t maxZ = halfLength, minZ = halfLength * -1.0f;
+       	float_t maxX = halfLength, minX = halfLength * -1.0f;
+
+		heightData = new float_t[(int) (gridSize * gridSize)]();
+		dim = (uint32_t) info -> m_GridSize;
+		dimScale = (uint32_t) info -> m_GridUnitSize;
+
+		Vector3 *vertexBuffer = m_Positions.GetArrayPtr();
+        uint32_t numVerts = m_Positions.Size();
+
+		uint32_t ibufLength;
+		uint32_t *ibufData;
+		GetIndexBuffer(ETopology::TriangleList, ibufLength, ibufData);
+
+		for (int i = 0; i < numVerts; i++)
+		{
+			heightData[i] = -1.0f;
+		}
+
+		for (int i = 0; i < (int) ibufLength; i++)
+		{
+			Vector3& curVert = vertexBuffer[ibufData[i]];
+
+			if (fmod(curVert.m_X, gridUnitSize) > .1 || //omit irregularities
+				fmod(curVert.m_Z, gridUnitSize) > .1)
+			{
+				continue;
+			}
+        	
```

Comment:
Holes will be marked with -1.0 in the heightmap (limited to 0.0-1.0 for normal values ofc) and irregularities in the vertex buffer are caught by the `fmod` checks.